### PR TITLE
Render religious landuse and place of worship lighter

### LINKED ISSUE #3322: Please add name rendering for landuse=religious
We have rendering of landuse since #2751
I moved the poi data from the building in the outline of https://www.openstreetmap.org/way/611970939
This is a parish centre.
Many current `name` I checked should be moved to `operator` however this would probably be done if the name is rendered.

### LINKED ISSUE #3457: Religious areas are rendered as massive building
### Expected behavior
They may have a color

### Actual behavior
But they look like a solid building

### Links and screenshots illustrating the problem
https://www.openstreetmap.org/way/621570785

screenshot_2018-10-17 linia 621570785 openstreetmap

### PR DESCRIPTION
**Pull request by jeisenbe**
Fixes #3457 
Fixes #3322

Changes proposed in this pull request:

- Change the fill color for religious landuse and places of worship to a lighter shade of gray

- Change the outline to 30% darker than the new shade; significantly lighter than the previous outline

Test rendering with links to the example places:

**Wamena, Indonesia**
http://openstreetmap.org/#map=17/-4.09339/138.94656
z15 Before:

After:

z16 Before

After:

z17 Before

After:

**YWAM campus in Kona, Hawaii**
(This is a large religious campus)
http://openstreetmap.org/#map=16/19.6287/-155.9750

z15 Before:

After:

z16 Before:
ywamz16original

After:

### CODE REVIEW

**Comment by kocio-pl** [comment]
Did you try lighter alternatives? I guess that it could be still less dark.

**Comment by jeisenbe** [comment]
Yeah, I don’t know why it is showing all of the previous commits in my
fork, even though I just created this branch last week. Should I rebase the
branch on master or will that cause more problems? Or squash the commits?

**Comment by jeisenbe** [comment]
Sorry for the delay, I meant to post these but my computer crashed.
Here are some renderings with #d6d6d6
I also tried lighter shades of gray, but I believe they are too close to residential.
This is about the limit, the lightest shade of gray that is still distinguishable:

z16 Wamena
religious-z16

z17 Wamena
religious-z17

z14 Kona
ywam-z14

z15 Kona
ywam-z15

z16 Kona
ywam-z16

**Comment by kocio-pl** [comment]
I guess this is good enough for me. Any other opinons?

**Comment by jragusa** [comment]
What about name rendering? For example, a lot of of these surfaces contain a name at Milano (see [overpass request](https://overpass-turbo.eu/s/Dr6))

**Comment by kocio-pl** [comment]
Also this Kona campus has name:

https://www.openstreetmap.org/way/101222222

It makes a lot of sense, on the other hand it should be tested if such names do not lead to clutter - I mean eclipsing symbols and names of churches/temples.

**Comment by kocio-pl** [comment]
BTW: `denomination=foursquare`? :smile: 

https://www.openstreetmap.org/way/607212018

**Comment by jeisenbe** [comment]
Good idea, I’ll try adding name labels.

(Foursquare, officially “The International Church of the Foursquare Gospel,
is a evangelical/protestant Christian denomination from the
Pentecostal/Charismatic tradition, founded by Amy Semple-McPherson in Los
Angeles in 1924; they “preach the foursquare Gospel.” Their logo has 4
symbols: a cross, crown, cup and dove. There is a big church in LA called
“Angeles Temple which used to be famous for extravagant Christmas pageants
and passion plays with live animals and flying “angels” thanks to
congregation members who worked in the Hollywood film business when Mrs.
Semple-McPherson was the preacher.

Probably they should have defended that trademark better...)

**Comment by jeisenbe** [comment]
Here are two options. 

**1) Add landuse_religious to the current rendering for amenity_place_of_worship,** 
plus a new line to allow lower zoom levels with waypixels >3000, as with other similar features

Kona z14
kona-z14-standard

Kona z15
kona-z16-standard

Kona z16
kona-z16-standard

**2) Change rendering to match that of other landuses and societal amenities**

Kona z14
kona-z14

Kona z15
kona-z15

Kona z16
kona-z16


**Code 1):**
```
  [feature = 'amenity_place_of_worship'],
  [feature = 'landuse_religious'] {
    [zoom >= 10][way_pixels > 3000][is_building = 'no'],
    [zoom >= 17] {
      text-name: "[name]";
      text-size: @standard-font-size;
      text-wrap-width: @standard-wrap-width;
      text-line-spacing: @standard-line-spacing-size;
      text-fill: #000033;
      text-dy: 12;
      text-face-name: @standard-font;
      text-halo-radius: @standard-halo-radius;
      text-halo-fill: @standard-halo-fill;
      text-placement: interior;
    }
  }
```
**Code 2):**

```
  [feature = 'amenity_place_of_worship'],
  [feature = 'landuse_religious'] {
    [zoom >= 10][way_pixels > 3000][is_building = 'no'],
    [zoom >= 17] {
      text-name: "[name]";
      text-size: @landcover-font-size;
      text-wrap-width: @landcover-wrap-width-size;
      text-line-spacing: @landcover-line-spacing-size;
      text-fill: #333333;
      text-dy: 12;
      [way_pixels > 12000] {
        text-size: @landcover-font-size-big;
        text-wrap-width: @landcover-wrap-width-size-big;
        text-line-spacing: @landcover-line-spacing-size-big;
      }
      [way_pixels > 48000] {
        text-size: @landcover-font-size-bigger;
        text-wrap-width: @landcover-wrap-width-size-bigger;
        text-line-spacing: @landcover-line-spacing-size-bigger;
      }
      text-face-name: @landcover-face-name;
      text-halo-radius: @standard-halo-radius;
      text-halo-fill: @standard-halo-fill;
      text-placement: interior;
    }
  }
```

**Comment by kocio-pl** [comment]
It would be interesting to see how it works with churches and other temples visible currently.

**Comment by kocio-pl** [comment]
Kona campus should be tagged as `amenity=university` by the way.

**Comment by jeisenbe** [comment]
Hope Chapel
1)Standard code
hopechapel-z16-standard
2)Landuse code
hopechapel-z16-landuse

Hawaiian Place of Worship (temple ruins)
1) Standard 
hawaiian-pow-z17-standard
2) Landuse
hawaiian-pow-z17-landuse

Bhuddist POW
1) Standard
mission-z17-standard

2) Landuse
mission-z17-landuse

**Comment by jeisenbe** [comment]
So I need to find a named landuse=religious polygon with multiple smaller
place_of_worship areas or nodes inside with different names.

How can I search for that?

Or I could invent one on JOSM

**Comment by jeisenbe** [comment]
Fixes #3322 

Amenity=place_of_worship labels will continue to be rendered the same as currently.

Landuse=religious name label will be rendered the same as other landuse labels.
 
I've done the renderings with #d0d0d0 as the color for the area, and darkened it 50% for the religious landuse label.

I added the names to some areas in my town on JOSM for testing

z17 Landuse=religious with 2 place_of_worship buildings and a node, in town.
z17-gkip-new-png

z18
z18-gkip-new

z19
z19-gkip-new

Church in the country (it's a round building)
z17
z17-katolik-new

z18
z18-katolik-new

Test with large religious area with place_of_worship building almost at the center, made to see how the conflicting labels would resolve. It behaves as expected.
z15
z15-kecandian-new

z16
z16-kecandian-new

z17
z17-kecandian-new

z18
z18-kecandian-new

File: amenity-points.mss [hunk 1]
```diff
@@ -1808,12 +1808,37 @@
...
+  }
+
+  [feature = 'amenity_place_of_worship'][zoom >= 16][way_pixels > 3000],
   [feature = 'amenity_place_of_worship'][zoom >= 17] {
     text-name: "[name]";
     text-size: @standard-font-size;
     text-wrap-width: @standard-wrap-width;
     text-line-spacing: @standard-line-spacing-size;
-    text-fill: #000033;
+    text-fill: #000033;
```
Comment (kocio-pl) [inline amenity-points.mss]: Just a space to remove at the end.

**Comment by jeisenbe** [comment]
It should be fixed now

**Comment by kocio-pl** [comment]
I'm happy with the results. Please just remove the extra space you have introduced by accident and this will be ready to merge - see https://github.com/gravitystorm/openstreetmap-carto/pull/3493#discussion_r231900882.

**Comment by polarbearing** [comment]
yes d0d0d0 is fine for me

**Comment by jeisenbe** [comment]
Should be fixed now.

It’s strange; I fixed it locally on my system and used $ git push, and it
says all the changes are pushed, but github didn’t show the file change.

So I fixed it on GitHub.

Please comment on the new github workflow issue if you have time.

**Comment by kocio-pl** [comment]
Thanks for the change and adding the label (thanks, @polarbearing!).

I'm not able to give more attention to the OSM Carto than I already do, sorry about that. I'm not sure what was the reason and it takes some time to investigate, but in the end the most important thing is that I can squash and merge your code properly.

**Pull request merged**